slsa-provenance/generator/Dockerfile

```diff
@@ -0,0 +1,14 @@
+# golang:1.17.1-alpine (linux/amd64)
+FROM golang@sha256:13919fb9091f6667cb375d5fdf016ecd6d3a5d5995603000d422b04583de4ef9 AS builder
+
+WORKDIR /app
+COPY . /app
+
+RUN go mod vendor
+RUN CGO_ENABLED=0 go build -ldflags="-w -s" -v -o app ./cmd
```

Comment:
this entire dockerfile can go into the action's `run` step. Can d that in follow-up PR